tests/unit/test_check_required_checks.py

```
@@ -0,0 +1,214 @@
+@pytest.mark.unit
+def test_fetch_branch_protection_contexts_handles_null_response(
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    """Repos with protection but no required_status_checks return null from --jq."""
+    import subprocess
+
+    class _FakeResult:
+        returncode = 0
+        stdout = "null\n"
+
+    def _fake_run(*args, **kwargs):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_fake_run`

(ANN202)